### PR TITLE
[RFC] docs(contributing): suggest following conventional commits

### PR DESCRIPTION
**Pull request by mjlbach**
@mhinz first brought this up to me ( I think with this gist https://gist.github.com/brianclements/841ea7bffdb01346392c) , but I wanted to update the contributing guidelines. Let me know if the feedback/examples are unclear or if we should add more additional guidelines.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,16 +74,26 @@ For Comment) and `[RDY]` (Ready).
...
-- **Prefix the commit subject with a _scope_:** `doc:`, `test:`, `foo.c:`,
+```
+<type>([optional scope]): <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+- **Prefix the commit subject with a _type_:** `doc:`, `test:`, `foo.c:`,
```
Comment (vigoux) [inline CONTRIBUTING.md:88]: I think the `foo.c` does not belong here anymore ?
```suggestion
- **Prefix the commit subject with a _type_:** `doc:`, `test:`
```

**Comment by clason** [comment]
Did you mean to commit the flake with this PR?

**Comment by mjlbach** [comment]
Nope! My bad

________________________________

**Comment by teto** [comment]
that was next in my todo. Great ! I would mention the part about `BREAKING CHANGE` too. Could potentially autocommit to https://github.com/neovim/neovim/issues/14090

**Comment by mjlbach** [comment]
@teto What do you think of the wording?

**Review by teto — APPROVED** [review]

**Comment by teto** [comment]
I reworded the commit message since it's not a suggestion but something we need to enforce. Could be mentioned in the breaking change issue even though it concerns just contributors.

**Pull request merged**